oauth.go

```diff
@@ -0,0 +1,80 @@
+package vrest
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"time"
+)
+
+var ErrOAuthTokenRequestFailed = errors.New("failed to get new oauth token")
+
+// OAuthConfig is the configuration for an OAuth token request.
+type OAuthConfig struct {
+	URL          string
+	GrantType    string
+	Scope        string
+	ClientID     string
+	ClientSecret string
+}
+
+// OAuthToken is an OAuth token.
+type OAuthToken struct {
+	AccessToken  string `json:"access_token"`
+	TokenType    string `json:"token_type"`
+	ExpiresIn    int    `json:"expires_in"`
+	ExtExpiresIn int    `json:"ext_expires_in"`
+	// ValidUntil is the time when the token is no longer valid.
+	// It is calculated when a new token is received.
+	ValidUntil time.Time `json:"-"`
+}
+
+type oauthTokenGetter struct {
+	config OAuthConfig
+	client *Client
+}
+
+// GetToken is a TokenGetter implementation that requests a new OAuth token.
+// No synchronization or locking is required in this function.
+func (o *oauthTokenGetter) GetToken(ctx context.Context, oldToken Token) (Token, error) {
+	body := url.Values{}
+	body.Add("grant_type", o.config.GrantType)
+	body.Add("scope", o.config.Scope)
+	body.Add("client_id", o.config.ClientID)
+	body.Add("client_secret", o.config.ClientSecret)
+
+	var token OAuthToken
+	err := o.client.NewRequestWithContext(ctx).
+		SetBaseURL(o.config.URL).
+		SetContentType("application/x-www-form-urlencoded").
+		SetBody(body.Encode()).
+		SetTokenRequest().
+		SetResponseBody(&token).
+		DoPost("")
+	if err != nil {
+		return nil, errors.Join(ErrOAuthTokenRequestFailed, err)
+	}
```

Comment:
```suggestion
	var token OAuthToken
	
	err := o.client.NewRequestWithContext(ctx).
		SetBaseURL(o.config.URL).
		SetContentType("application/x-www-form-urlencoded").
		SetBody(body.Encode()).
		SetTokenRequest().
		SetResponseBody(&token).
		DoPost("")
	if err != nil {
		return nil, errors.Join(ErrOAuthTokenRequestFailed, err)
	}
```

Cuddling, we need to set up `golangci-lint` here

Reply:
This cuddling makes sense here, because this whole block belongs together.
There is also a disagreement between gofumpt and the wsl linter. wsl complains about cuddling, and gofumpt always cuddles these blocks. I agree with gofumpt here.

There is also a guthub issue with a discussion on this topic:
https://github.com/golangci/golangci-lint/issues/1510

I think we have to disable the wsl linter.

Reply:
I haven't found a way to disable the check that leads to the warning "only one cuddle assignment allowed before if statement (wsl)".